packages/cli/src/databases/repositories/execution.repository.ts

```diff
@@ -399,21 +405,89 @@ export class ExecutionRepository extends Repository<ExecutionEntity> {
 
 		if (!executions.length) {
 			if (deleteConditions.ids) {
-				LoggerProxy.error('Failed to delete an execution due to insufficient permissions', {
+				Logger.error('Failed to delete an execution due to insufficient permissions', {
 					executionIds: deleteConditions.ids,
 				});
 			}
 			return;
 		}
 
 		const executionIds = executions.map(({ id }) => id);
-		const binaryDataManager = BinaryDataManager.getInstance();
-		await binaryDataManager.deleteBinaryDataByExecutionIds(executionIds);
-
 		do {
 			// Delete in batches to avoid "SQLITE_ERROR: Expression tree is too large (maximum depth 1000)" error
-			const batch = executionIds.splice(0, 500);
-			await this.delete(batch);
+			const batch = executionIds.splice(0, this.deletionBatchSize);
+			await this.softDelete(batch);
 		} while (executionIds.length > 0);
 	}
+
+	async pruneBySoftDeleting() {
+		Logger.verbose('Pruning (soft-deleting) execution data from database');
+
+		const maxAge = config.getEnv('executions.pruneDataMaxAge'); // in h
+		const maxCount = config.getEnv('executions.pruneDataMaxCount');
+
+		// Find ids of all executions that were stopped longer that pruneDataMaxAge ago
+		const date = new Date();
+		date.setHours(date.getHours() - maxAge);
+
+		const toPrune: Array<FindOptionsWhere<IExecutionFlattedDb>> = [
+			// date reformatting needed - see https://github.com/typeorm/typeorm/issues/2286
+			{ stoppedAt: LessThanOrEqual(DateUtils.mixedDateToUtcDatetimeString(date)) },
```

Comment:
The problem of filtering by `stoppedAt` is that I think this value might be `null` for errored executions and also maybe for crashed ones. It's something worth checking.

If we're sure this column is always filled (and it should be IMO), then we can use this value alone to performa a single direct update as well

Reply:
Definitely a good point! I only wonder if we should be tackling this now? Improving the existing pruning logic was [descoped](https://linear.app/n8n/issue/PAY-771/implement-soft-deletion) to shorten this big tech debt detour, so I can start on the S3 implementation. Perhaps something to address separately? Unless it's high priority.

Reply:
It's ok, we can do it separately, doesn't have to be in the scope of this PR.

Reply:
Added story to backlog: https://linear.app/n8n/issue/PAY-823/ensure-no-null-values-when-filtering-by-stoppedat-to-prune-executions